_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,276 @@
+---
+id: efbfc7
+name: Text content that updates automatically can be paused, stopped or hidden
+rule_type: atomic
+description: |
+  This rule checks that there are mechanisms to pause, stop or hide the auto-updating of text content.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:2.2.2: # Pause, Stop, Hide (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [visible][] [HTML element][] in an [HTML document][] if:
+
+- the `innerText` property of the [element][html element] changes;
+- the [element][html element] does not have [children][child] whose `innerText` property is also changed;
+- the change happens after the [readiness][document readiness] of the [HTML document][] the [element][html element] belongs to is equal to "complete";
+- the change happens when the [last activation timestamp][] of the [window][] the [HTML document][] is associated with has the value "positive infinity";
+- it is not the only [content][] in the [HTML document][].
+
+## Expectation 1
+
+For the test target a [user interface component][] is provided to pause, stop, hide or alter the frequency of the changes of the [visible text content][].
```

Comment:
If the stop mechanism is a global shortcut, this will fail the rule but pass the SC…

```html
<body onload="startUpdates()">
  <script type="text/javascript" src="/test-assets/efbfc7/script.js"></script>
  <p>Random number: <span id="target">1</span></p>
  <p>Press "+" to stop updates"</p>
  <script>
    document.body.addEventListener("keydown", function(event) {
      if (event.key === "+") { stopUpdates(); }
    }
  </script>
</body>
```

Reply:
You are right. The only way I can see to address this is to use [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) similarly to what was done in [audio or video that plays automatically has a control mechanism](https://act-rules.github.io/rules/4c31df)

Reply:
Yes. Problem is that we then need an unambiguous definition of mechanism...
#1016